meta/data/CorrectCifarDataset.py

```diff
@@ -198,6 +198,83 @@ def increase_tasks_and_data_per_task(self, task_increase, data_increase):
         if task_increase > 0:
             self.add_new_tasks(task_increase, data_for_new_tasks//self.classes_per_task) # debugged by wangq 20/12/2020
 
+class CifarStaticDatasetHierarchy(CifarStaticDataset):
+    '''
+    Implementation of a static Cifar Dataset that takes into account the data structure to create hard and easy tasks
+    '''
+    def __init__(self, root_dir, mode, hierarchy_json, no_of_easy, no_of_hard, classes_per_task, no_data_points_hard, no_data_points_easy):
+        '''
+        root_dir: directory containing the data. CAREFUL: If mode is Mix you would like to use diferent data directories.
+        no_of_easy: positive integer - initial number of easy tasks to sample.
+        no_of_data_points_per_task - positive int.
+        root_dir - data folder
+        mode - string 'Train', 'Test','Val', 'Mix_Train','Mix_Test', 'Mix_Val'. 
+        With 'Mix' prefix clases are share between, train, test and validation  
+        '''
+
+        self.classes_per_task = classes_per_task
+        self.task_additions = 0
+        self.task_seed = 0 ## ?what is this for? wherent you seeding at main?
+        
+        self.classes_per_task = classes_per_task
+        with open(hierarchy_json) as json_file: 
+            data = json.load(json_file) 
+        
+        self.hyperclass_strucutre = data[mode] if 'Mix' not in mode else data['Mix'] 
+        
+        hard_tasks_list = []
+        _tasks_classes = []
+        for key, value in self.hyperclass_strucutre.items() :
+            hard_tasks_list +=list(itertools.combinations(value, self.classes_per_task))
+            _tasks_classes += value
+        
+        all_tasks_list = list(itertools.combinations(_tasks_classes, self.classes_per_task))
+        easy_tasks_list = list (set(all_tasks_list)-set (hard_tasks_list) )
+    
+        #This shuffle needs a seed, is random seeded on main?
+        random.shuffle( hard_tasks_list)
```

Comment:
Let's move this shuffling to the generate task parametrization function. It logically belongs there.